### PR TITLE
replace TestBase#assertContains(String sub, String string) with Asser…

### PR DESCRIPTION
**Pull request by PascalSchumacher**
…tJ #contains and #hasMessageContaining

### CODE REVIEW

File: src/test/java/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNullsTest.java [hunk 1]
```diff
@@ -51,8 +53,9 @@ public void should_return_an_object_that_allows_object_methods() throws Throwabl
 
         Foo smartNull = (Foo) answer.answer(invocationOf(Foo.class, "get"));
 
-        assertContains("SmartNull returned by", smartNull + "");
-        assertContains("foo.get()", smartNull + "");
+        assertThat(smartNull.toString())
```
Comment (szpak) [inline src/test/java/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNullsTest.java]: I wonder if a risk of NPE is real in those tests. With `smartNull + ""` the nicer error message would be displayed when `smartNull` is `null`.

File: src/test/java/org/mockitousage/stacktrace/ClickableStackTracesTest.java [hunk 2]
```diff
@@ -34,8 +35,7 @@ public void shouldShowActualAndExpectedWhenArgumentsAreDifferent() {
             verifyTheMock(1, "not foo");
             fail();
         } catch (ArgumentsAreDifferent e) {
-            assertContains("callMethodOnMock(", e.getMessage());
-            assertContains("verifyTheMock(", e.getMessage());
+            assertThat(e).hasMessageContaining("callMethodOnMock(").hasMessageContaining("verifyTheMock(");
```
Comment (szpak) [inline src/test/java/org/mockitousage/stacktrace/ClickableStackTracesTest.java]: To keep consistency the second `hasMessageContaining` should be in the another line.

File: src/test/java/org/mockitousage/verification/DescriptiveMessagesWhenVerificationFailsTest.java [hunk 3]
```diff
@@ -138,8 +138,7 @@ public void should_print_actual_and_wanted_when_actual_method_name_and_wanted_me
             verify(mock).simpleMethod(10);
             fail();
         } catch (ArgumentsAreDifferent e) {
-            assertContains("simpleMethod(10)", e.getMessage());
-            assertContains("simpleMethod()", e.getMessage());
+            assertThat(e).hasMessageContaining("simpleMethod(10)").hasMessageContaining("simpleMethod()");
```
Comment (szpak) [inline src/test/java/org/mockitousage/verification/DescriptiveMessagesWhenVerificationFailsTest.java]: The same here.

**Comment by szpak** [comment]
LGTM.

It was a very ungrateful work, thanks Pascal!

File: src/test/java/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNullsTest.java (hunk 1 above)
Comment (bric3) [inline src/test/java/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNullsTest.java]: For me that's ok, the NPE is straightforward in this case.

**Pull request merged**